tests/plottable.py

```diff
@@ -0,0 +1,154 @@
+#     COUNT = "COUNT"
+#     MEAN = "MEAN"
+# Then return and use Kind.COUNT or Kind.MEAN.
+
+
+@runtime_checkable
+class PlottableTraits(Protocol):
+    # True if the axis "wraps around"
+    circular: bool
+
+    # True if each bin is discrete - Integer, Boolean, or Category, for example
+    discrete: bool
+
+
+T = TypeVar("T", covariant=True)
+
+
+@runtime_checkable
+class PlottableAxisGeneric(Protocol[T]):
+    # name: str - Optional, not part of Protocol
+    # label: str - Optional, not part of Protocol
+    #
+    # Plotters are encouraged to plot label if it exists and is not None, and
+    # name otherwise if it exists and is not None, but these properties are not
+    # available on all histograms and not part of the Protocol.
+
+    traits: PlottableTraits
+
+    def __getitem__(self, index: int) -> T:
+        """
+        Get the pair of edges (not discrete) or bin label (discrete).
+        """
+
+    def __len__(self) -> int:
+        """
+        Return the number of bins (not counting flow bins, which are ignored
+        for this Protocol currently).
+        """
+
+    def __eq__(self, other: Any) -> bool:
+        """
+        Required to be sequence-like.
+        """
+
+
+PlottableAxisContinuous = PlottableAxisGeneric[Tuple[float, float]]
+PlottableAxisInt = PlottableAxisGeneric[int]
+PlottableAxisStr = PlottableAxisGeneric[str]
+
+PlottableAxis = Union[PlottableAxisContinuous, PlottableAxisInt, PlottableAxisStr]
+
+
+@runtime_checkable
+class PlottableHistogram(Protocol):
+    axes: Sequence[PlottableAxis]
+
+    kind: Kind
+
+    def values(self) -> ArrayLike:
+        """
+        Returns the accumulated values. The counts for simple histograms, the
+        sum of weights for weighted histograms, the mean for profiles, etc.
+
+        If counts is equal to 0, the value in that cell is undefined if
+        kind == "MEAN".
+
+        All methods can have a flow=False argument - not part of this Protocol.
+        If this is included, it should return an array with flow bins added,
+        normal ordering.
+        """
+
+    def variances(self) -> Optional[ArrayLike]:
+        """
+        Returns the estimated variance of the accumulated values. The sum of squared
+        weights for weighted histograms, the variance of samples for profiles, etc.
+        For an unweighed histogram where kind == "COUNT", this should return the same
+        as values if the histogram was not filled with weights, and None otherwise.
+
+        If counts is equal to 1 or less, the variance in that cell is undefined if
+        kind == "MEAN".
+
+        If kind == "MEAN", the counts can be used to compute the error on the mean
+        as sqrt(variances / counts), this works whether or not the entries are
+        weighted if the weight variance was tracked by the implementation.
+        """
+
+    def counts(self) -> Optional[ArrayLike]:
+        """
+        Count returns the number of entries in each bin for an unweighted
```

Comment:
Don't repeat the docstring here. They will get out of sync.

Reply:
This is just an vendored copy of the file from `henryiii/uhi` so that we don't have to depend on something in my namespace that isn't published to PyPI yet. Once I clean up UHI, it will be a repo that helps with testing UHI implementations, and will include these protocols for testing PlottibleProtocol Produces and Consumers, we can drop this file and just use this (testing requirement only).

That's why it's a duplicate, though - it's only put here for testing purposes.